### DISCUSSION TITLE
Use `pnpm install --shamefully-hoist` to install dependencies

### DISCUSSION THREAD

**Discussion by kuizuo** [nuxt/content#1602] [General]
When you use pnpm as package management, if you use `pnpm install` to install dependencies. The following error may appear.

```
 WARN  Failed to resolve dependency: vue, present in 'optimizeDeps.include'                                                           01:18:16


 WARN  Failed to resolve dependency: html-tags, present in 'optimizeDeps.include'                                                     01:18:16


 ERROR  The following dependencies are imported but could not be resolved:                                                            01:18:16

  vue (imported by F:/Demo/nuxt-app/node_modules/.pnpm/nuxt@3.0.0-rc.11/node_modules/nuxt/dist/app/entry.mjs)

Are they installed?

i Vite client warmed up in 70ms                                                                                                       01:18:16
√ Nitro built in 1254 ms
```

At this time node_modules structure is as follows:

![QQ截图20221016012304](https://user-images.githubusercontent.com/61005888/195999769-1b968c45-63e7-4195-aad7-c74384474234.png)

This leads to `Failed to resolve dependency`, Can be used `pnpm install --shamefully-hoist` to create a flat node_modules directory structure, similar to npm or yarn. This can effectively solve the above problems.

![image](https://user-images.githubusercontent.com/61005888/195999890-5a277da1-1690-4f94-87e2-61570aaadad0.png)

**Comment by nobkd** [comment]
How to install the project with pnpm can also be seen on the [get started page](https://content.nuxtjs.org/get-started) when selecting pnpm option
![Screenshot of the pnpm console command: pnpm install --shamefully-hoist](https://user-images.githubusercontent.com/44443899/196023225-f6d67dbd-5efc-4238-8dfa-3c8aef926480.png)
